doc/website/release-notes/iceoryx-unreleased.md

```diff
@@ -400,3 +401,23 @@
     // after
     #include "iceoryx_platform/some_header.hpp"
     ```
+
+22. `cxx::unique_ptr` is no longer nullable.
+
+    ```cxx
+    // before
+    cxx::unique_ptr<int> myPtr(ptrToInt, someDeleter);
+    cxx::unique_ptr<int> emptyPtr(nullptr, someDeleter);
+
+    myPtr.reset(ptrToOtherInt);
+    myPtr.release();
+
+
+    // after
+    cxx::unique_ptr<int> myPtr(*ptrToInt, someDeleter); // requires reference to object to ensure non nullptr
+    cxx::optional<cxx::unique_ptr<int>> emptyPtr(cxx::nullopt); // if unique_ptr shall be nullable use cxx::optional
+
+    myPtr.reset(*ptrToOtherInt); // requires reference to object to ensure non nullptr
+    cxx::unique_ptr<int>::release(std::move(myPtr)); // release consumes myPtr
+
+    ```
```

Comment:
Could you maybe document somewhere that gcc/clang will issue a warning if a moved-from object is used and since there is no other way the unique_ptr can be invalid, it also does not need to be checked for null?